src/lib.rs

```diff
@@ -105,6 +166,47 @@ impl DstService {
         Ok(())
     }
 
+    fn is_add(prev: &HashMap<SocketAddr, EndpointMeta>, ep: &EndpointMeta) -> bool {
+        match prev.get(&ep.address) {
+            Some(prev_ep) => prev_ep != ep,
+            None => true,
+        }
+    }
+
+    fn to_weighted_addr(meta: &EndpointMeta) -> pb::WeightedAddr {
```

Comment:
Nit/TIOLI: this seems like it should probably be a method on `EndpointMeta`, rather than an associated function on `DstService`?